### PR TITLE
Docs review of api_app.md

### PR DESCRIPTION
**Pull request by maclover7**
Pass through correcting api_app.md. The list of included modules and
middleware was tested through a sample API app, and was listed in the
same order an end user would see in their terminal.

[ci skip]

### CODE REVIEW

**Comment by maclover7** [comment]
ref #23322

File: guides/source/api_app.md [hunk 1]
```diff
@@ -360,26 +361,28 @@ Choosing Controller Modules
 An API application (using `ActionController::API`) comes with the following
 controller modules by default:
 
-- `ActionController::UrlFor`: Makes `url_for` and friends available.
+- `ActionController::UrlFor`: Makes `url_for` and similar helpers available.
 - `ActionController::Redirecting`: Support for `redirect_to`.
-- `ActionController::Rendering`: Basic support for rendering.
```
Comment (rafaelfranca) [inline guides/source/api_app.md]: `ActionController::Rendering` is also included.

File: guides/source/api_app.md [hunk 2]
```diff
@@ -360,26 +361,28 @@ Choosing Controller Modules
...
+- `ActionController::Rescue`: Support for `rescue_from`.
+- `ActionController::Instrumentation`: Support for the instrumentation
+  hooks defined by Action Controller (see [the instrumentation
+  guide](active_support_instrumentation.html#action-controller) for
+more information regarding this).
 - `ActionController::ParamsWrapper`: Wraps the parameters hash into a nested hash
   so you don't have to specify root elements sending POST requests for instance.
 
-Other plugins may add additional modules. You can get a list of all modules
+Other gems or plugins may add additional modules. You can get a list of all modules
```
Comment (rafaelfranca) [inline guides/source/api_app.md]: `plugins` include gems.

File: guides/source/api_app.md [hunk 3]
```diff
@@ -208,7 +209,7 @@ An API application comes with the following middleware by default:
 See the [internal middleware](rails_on_rack.html#internal-middleware-stack)
 section of the Rack guide for further information on them.
 
-Other plugins, including Active Record, may add additional middleware. In
+Other gems or plugins, including Active Record, may add additional middlewares. In
```
Comment (rafaelfranca) [inline guides/source/api_app.md]: plugins includes `gems`.

Comment (rafaelfranca) [inline guides/source/api_app.md]: `middleware`

**Comment by maclover7** [comment]
@rafaelfranca updated.

**Pull request merged**